### PR TITLE
Refactor documentation

### PR DESCRIPTION
**Pull request by dg-concordium**
## Purpose

Remove outdated information. Update wording, add more glossary links, etc.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [x] I accept the above linked CLA.

### CODE REVIEW

File: source/mainnet/net/mobile-wallet/update-baker-mw.rst
Comment (jens-concordium) [inline source/mainnet/net/mobile-wallet/update-baker-mw.rst:61]:
```suggestion
When you change the stake it can influence your :ref:`chance of being selected to bake a block<glossary-winning-probability>` and receive baker rewards and of being included in the :ref:`finalization <glossary-finalization>` committee. If you decrease the stake, you decrease your chances of baking blocks and of being included in the finalization committee. Likewise, if you increase the stake, you increase your chances of baking a block and of being included in the finalization committee.
```

File: source/mainnet/net/resources/glossary.rst
Comment (jens-concordium) [inline source/mainnet/net/resources/glossary.rst:470]: This section sounds correct to me, but it would probably be good to have an extra set of eyes on it. Do you have any comments on this description of private keys, @abizjak?

File: source/mainnet/net/concepts/concepts-baker.rst
Comment (dg-concordium) [reply source/mainnet/net/concepts/concepts-baker.rst:156]: It is because I repurposed the old topic name instead of renaming the file and having to correct a lot of cross references. The reader does not see it unless they look at the path in the URL.

File: source/mainnet/net/guides/export-import.rst
Comment (dg-concordium) [reply source/mainnet/net/guides/export-import.rst:12]: For Desktop Wallet yes. It is also Wallet for the other wallet names in the variables file.

File: source/mainnet/net/mobile-wallet/update-baker-mw.rst
Comment (dg-concordium) [reply source/mainnet/net/mobile-wallet/update-baker-mw.rst:43]: We've never had a screenshot for this. Is it the same as for gen1? Otherwise, I will need to get an Android phone and take one.

Comment (jens-concordium) [reply source/mainnet/net/mobile-wallet/update-baker-mw.rst:43]: Yep, same as gen1. It looked like there was meant to be one in the "code" as well, but it just didn't render in the website

File: source/mainnet/net/guides/export-import.rst
Comment (jens-concordium) [reply source/mainnet/net/guides/export-import.rst:12]: Alright!

**Comment by dg-concordium** [comment]
@jens-concordium Update baker screenshot is added. Can you review again?

File: source/mainnet/net/resources/glossary.rst
Comment (jens-concordium) [inline source/mainnet/net/resources/glossary.rst:491]: Was this just added now? I didn't see it before. 

It makes perfect sense to me, but I have no idea whether the definition is correct or not

Comment (dg-concordium) [reply source/mainnet/net/resources/glossary.rst:491]: Ales has already reviewed Qualified authority. https://github.com/Concordium/concordium.github.io/pull/745

File: source/mainnet/net/resources/glossary.rst
Comment (jens-concordium) [reply source/mainnet/net/resources/glossary.rst:484]: I resolved it this, just so I'm not blocking anything while I'm away the next two days. Daniel will wait for your approval before merging it, Ales.

**Review by jens-concordium — APPROVED** [review]
Approved, but waiting for the final go from Ales on the private key thing.

**Pull request merged**
